accounts/modelbackend.py

```diff
@@ -23,11 +23,32 @@
 
 
 class CustomModelBackend(ModelBackend):
-    def authenticate(self, username=None, password=None):
+
+    def authenticate(self, request, username=None, password=None, **kwargs):
         """ authenticate against case insensitive username """
-        try:
-            user = User.objects.get(username__iexact=username)
-            if user.check_password(password):
-                return user
-        except User.DoesNotExist:
-            return None
+
+        if '@' in username:
```

Comment:
I don't really understand all of this if/else stuff, especially as the login via username is in here twice (DRY!).

Can't you just do the try/except and default to username first?

```python
user = None
try:
   user = User.objects.get(username__iexact=username) # default to username
except User.DoesNotExist:
   try:
       user = User.objects.get(email__iexact=username) # try with email
   except User.DoesNotExist:
       pass

if user and user.check_password(password):
   return user

return None
```